### PR TITLE
mcap futures settlement

### PR DESCRIPTION
**Pull request by davidsiska-vega**

### CODE REVIEW

File: notes/mcap-futures-markets-settlement.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Research note: settling market capitalisation pre-launch token futures
+
+This note is a response to community interest in creating futures markets on the market capitalisation of various tokens prior to token launch.
+
+It describes one way to define when and how to settle such markets given the broad possible number of outcomes and the lack of ahead-of-time certainty as to whether a token will launch and if it does, what the market capitalisation will be at launch.
+
+The approach is modelled on Credit Default Swaps (CDS) which are derivatives that are valued based on the probability of a company defaulting. Payouts are triggered when a “credit event” occurs, as defined by a set of explicit rules. Analogously, we define the concept of an “airdrop event” that can be similarly used to determine when and at what value to settle points futures.
```
Comment (witgaw) [inline notes/mcap-futures-markets-settlement.md:7]:
```suggestion
The approach is modelled on Credit Default Swaps (CDS) which are derivatives that are valued based on the probability of a company defaulting. Payouts are triggered when a “credit event” occurs, as defined by a set of explicit rules. Analogously, we define the concept of an “token launch” that can be similarly used to determine when and at what value to settle MCAP futures.
```

**Review by witgaw — COMMENTED** [review]
LGTM, just some minor suggestions:

File: notes/mcap-futures-markets-settlement.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+: *If specified* the market would settle at a price of zero at the expiry date/time if no accepted value has been produced by the ***MCAP Valuation Algorithm*** on or before the specified date and time.
+
+`Liquidity Threshold` (0.05% = 5bp = 0.0005)
+: The fraction of total number of tokens that must available to be bought *and* sold in the market (across valid exchanges defined below) within `valid slippage` (%) of the current price per point.
+
+`Valid Slippage` (20.00%)
+: The range around the current market price at which available bids are deemed to constitute ‘liquidity’ for the purpose of the ***MCAP Valuation Algorithm***.
+
+`Minimum Trading Window` (48 hours)
+: The minimum amount of cumulative time over which a market meeting the liquidity and slippage thresholds must exist to trigger a succesful ***MCAP Valuation Algorithm*** calculation.
```
Comment (TomMcL) [inline notes/mcap-futures-markets-settlement.md:57]:
```suggestion
: The minimum amount of cumulative time over which a market meeting the liquidity and slippage thresholds must exist to trigger a successful ***MCAP Valuation Algorithm*** calculation.
```

**Review by TomMcL — COMMENTED** [review]
I wonder if it's worth adding a section about the possibility of a market amendment to align with actual number of tokens issued if it's announced?

**Review by barnabee — CHANGES_REQUESTED** [review]

File: notes/mcap-futures-markets-settlement.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+This note is a response to community interest in creating futures markets on the market capitalisation of various tokens prior to token launch, aka "MCAP futures" or "MCAP markets".
+
+It describes one way to define when and how to settle such markets given the broad possible number of outcomes and the lack of ahead-of-time certainty as to whether a token will launch and if it does, what the market capitalisation will be at launch.
+
+The approach is modelled on Credit Default Swaps (CDS) which are derivatives that are valued based on the probability of a company defaulting. Payouts are triggered when a “credit event” occurs, as defined by a set of explicit rules. Analogously, we define the concept of an “token launch” that can be similarly used to determine when and at what value to settle MCAP futures.
+
+
+## Conceptual overview
+
+Conceptually, we need to define three things to enable fair and orderly settlement of an mcap futures market:
```
Comment (davidsiska-vega) [inline notes/mcap-futures-markets-settlement.md:12]:
```suggestion
Conceptually, we need to define four things to enable fair and orderly settlement of an mcap futures market:
```

File: notes/mcap-futures-markets-settlement.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+This note is a response to community interest in creating futures markets on the market capitalisation of various tokens prior to token launch.
+
+It describes one way to define when and how to settle such markets given the broad possible number of outcomes and the lack of ahead-of-time certainty as to whether a token will launch and if it does, what the market capitalisation will be at launch.
+
+The approach is modelled on Credit Default Swaps (CDS) which are derivatives that are valued based on the probability of a company defaulting. Payouts are triggered when a “credit event” occurs, as defined by a set of explicit rules. Analogously, we define the concept of an “token launch” that can be similarly used to determine when and at what value to settle MCAP futures.
+
+
+## Conceptual overview
+
+Conceptually, we need to define three things to enable fair and orderly settlement of an mcap futures market:
```
[the remark this answers is not in the document]
Comment (davidsiska-vega) [reply notes/mcap-futures-markets-settlement.md]: Yea, fixed.

File: notes/mcap-futures-markets-settlement.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+
+### Variables
+
+The following terms (with suggested values in brackets) used in the definitions refer to quantities or other parameters that affect the specifics of the product. They may be varied to suit each individual use case.
+
+`Base Asset` (USDT)
+: The asset in which market capitalisation of the token is considered.
+
+`Inclusivity Threshold` (40.00%)
```
[the remark this answers is not in the document]
Comment (davidsiska-vega) [reply notes/mcap-futures-markets-settlement.md]: Changed.

File: notes/mcap-futures-markets-settlement.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+`Minimum Trading Window` (48 hours)
+: The minimum amount of cumulative time over which a market meeting the liquidity and slippage thresholds must exist to trigger a succesful ***MCAP Valuation Algorithm*** calculation.
+
+`Valid Exchanges`
+: A list of exchanges on which the listing of the token would count as valid for measurement of price. These may include a list of major centralised exchanges such as Binance, ByBit or Coinbase or significant decentralised spot exchanges on the chains where the token is likely to be launched (Uniswap for Ethereum, Osmosis for Cosmos etc.).
+
+
+### MCAP Valuation Algorithm
+
+Once the token has launched and has been trading for `Minimum Trading Window` with sufficient `Liquidity Threshold` the valuation algorithm will calculate the MCAP as the time-weighted average price, over the minimum trading window times the total number of tokens.
```
[the remark this answers is not in the document]
Comment (davidsiska-vega) [reply notes/mcap-futures-markets-settlement.md]: Re-wrote.

File: notes/mcap-futures-markets-settlement.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+`Expiry Date/Time` (optional)
+: *If specified* the market would settle at a price of zero at the expiry date/time if no accepted value has been produced by the ***MCAP Valuation Algorithm*** on or before the specified date and time.
+
+`Liquidity Threshold` (0.05% = 5bp = 0.0005)
+: The fraction of total number of tokens that must available to be bought *and* sold in the market (across valid exchanges defined below) within `valid slippage` (%) of the current price per point.
+
+`Valid Slippage` (20.00%)
+: The range around the current market price at which available bids are deemed to constitute ‘liquidity’ for the purpose of the ***MCAP Valuation Algorithm***.
+
+`Minimum Trading Window` (48 hours)
```
[the remark this answers is not in the document]
Comment (davidsiska-vega) [reply notes/mcap-futures-markets-settlement.md]: Changed to a week.

**Review by barnabee — APPROVED** [review]

**Pull request merged**